app/Http/Controllers/CategoryController.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Http\Resources\CategoryResource;
+use App\Models\Category;
+use Illuminate\Http\Request;
+use App\Http\Requests\CategoryRequest;
+
+
+class CategoryController extends Controller
+{
+    public function index(Request $request)
+    {
+        $perPageOptions = [20, 50, 100];
+
+        $perPage = $request->input('per_page', 20);
+
+        if (!in_array($perPage, $perPageOptions)) {
```

Comment:
@Geri-PG Can you change the logic to use validation for the per_page query parameter? If the query param is not set, the default value should be 20. Check the following validation rule [https://laravel.com/docs/11.x/validation#rule-in](url)